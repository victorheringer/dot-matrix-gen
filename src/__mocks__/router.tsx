import { MemoryRouter } from "react-router-dom";

export function RouterWrapper({ children }: any) {
  return <MemoryRouter>{children}</MemoryRouter>;
}
