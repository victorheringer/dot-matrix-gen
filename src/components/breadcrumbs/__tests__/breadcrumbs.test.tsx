import { RouterWrapper } from "../../../__mocks__/router";
import { render } from "@testing-library/react";
import { Breadcrumbs } from "../index";
import useBreadcrumbs from "use-react-router-breadcrumbs";

jest.mock("use-react-router-breadcrumbs");

describe("Breadcrumbs", () => {
  beforeAll(() => {
    (useBreadcrumbs as jest.Mock).mockReturnValue([
      { match: { pathname: "test" }, breadcrumb: "text-value" },
      { match: { pathname: "test-other" }, breadcrumb: "text-other-value" },
    ]);
  });

  it("should render correctly", () => {
    const tree = render(
      <RouterWrapper>
        <Breadcrumbs />
      </RouterWrapper>
    ).asFragment();
    expect(tree).toMatchSnapshot();
  });
});
