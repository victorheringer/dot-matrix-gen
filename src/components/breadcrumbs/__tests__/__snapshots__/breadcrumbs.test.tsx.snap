// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ fUFrXq"
  >
    <span>
      <a
        href="/test"
      >
        text-value
      </a>
       &gt; 
    </span>
    <span>
      <a
        href="/test-other"
      >
        text-other-value
      </a>
       
    </span>
  </div>
</DocumentFragment>
`;
